File: Oidc.AuthorizationServer/Models/CorsOrigin.cs
using lvl.Ontology;
using lvl.Ontology.Authorization;
using lvl.Ontology.Conventions;
using System.ComponentModel.DataAnnotations;

namespace lvl.Oidc.AuthorizationServer.Models
{
    /// <summary>
    ///     A site which is considered safe.
    /// </summary>
    [Schema("oidc")]
    [HiddenFromApi]
    public class CorsOrigin : Entity, IAggregateScope<ClientEntity>
    {
        /// <summary>
        ///     A url to a site.
        /// </summary>
        [Required]
        public string Name { get; set; }
    }
}

File: oidc/src/models/cors-origin.ts
import { ClientEntity } from './client-entity';
import { Entity, IAggregateScope, Required } from '@lvl/front-end';

export class CorsOrigin extends Entity implements IAggregateScope<ClientEntity> {
    @Required() public name: string;

    constructor(options?: ICorsOriginOptions) {
        options = options || {};
        super(options);
        this.name = options!.name;
    }
}

interface ICorsOriginOptions {
    name?: string;
    id?: number;
}
